cpp/src/aztec/plonk/proof_system/prover/c_bind.cpp

```diff
@@ -30,97 +30,100 @@ WASM_EXPORT void* test_async_func(size_t size, int val)
     }
 }
 
-WASM_EXPORT void prover_process_queue(waffle::TurboProver* prover)
+typedef std::conditional_t<waffle::SYSTEM_COMPOSER == waffle::TURBO, waffle::TurboProver, waffle::UltraProver>
```

Comment:
This prover should not be a compile time but runtime declaration. Use polymorphism (assign someone, should be done in a fresh PR)